features/karaf/openhab-core/src/main/feature/feature.xml

```diff
@@ -181,6 +181,8 @@
 
 	<feature name="openhab-core-io-websocket" version="${project.version}">
 		<feature>openhab-core-base</feature>
+		<feature dependency="true">openhab-core-io-rest-auth</feature>
```

Comment:
You can stick with `openhab-core-base` if you decide to go with `AuthenticationManager`, cause it will be provided earlier than rest filter.

Reply:
I think this is a good point and the `AuthenticationManager` is the way to go.

Reply:
This can not be done right now, because the JWT verification and the APISecurity options are both handled in the openhab.core.io.rest.auth package instead than in the core.

@splatch mentioned in another comment that is better to let that change for a future reorganization, which seems better for me, but let me know if this comment still applies.

Reply:
Agree, there is more than just `AuthenticationManager` to do, thus as long as we avoid leakage of `AuthFilter` logic beyond it it should be fine to do further updates after 4.0.